### ISSUE TITLE
[Feature] example for payment handling

### ISSUE THREAD

**Issue by niklaswolf** [shopware/frontends#33]
Right now, neither in the demo store template nor in the docs there is an example how to handle payments.
As this is a very crucial part of an online shop ( 😛 ), there should be an example on how to handle it. 

We had a similar issue in shopware-pwa, where I described what the flow could be (and that's also the flow how it was implemented in shopware-pwa, which was quite robust)
https://github.com/vuestorefront/shopware-pwa/issues/1419

**Comment by elkmod** [comment]
We are currently reviewing a guide for payment handling in our docs - which deals with the PayPal Express integration (as an example) specifically. I will post here, once it's merged.

**Comment by niklaswolf** [comment]
I think it's fine, even if it is really tightly coupled to the PayPal express integration. But because this is made quite clear in the beginning, I think it's fine.

Honestly, I was more hoping for a guide which explained the usage of the plain Shopware interfaces/api for payments (sync and async), because these are well defined and every payment provider should use these and provide additional APIs if necessary (like in your PayPal example). But these additional APIs have to be documented by the payment provider itself, when and how to use it. E.g. most of the Mollie async payment methods work fine just using the standard Shopware apis.

So if you really want to make this as transparent as possible, I'd encourage you to document that, too, and keep the PayPal example as an example of a more complex integration.

**Comment by elkmod** [comment]
I agree with @niklaswolf here - the specific example is good for the specific case, but if there's a "general" approach. I think what Niklas is referring to is something more similar to the [Payment API](https://shopware-pwa-docs.vuestorefront.io/landing/concepts/payment.html) section from our shopware-pwa docs. We could then move the PayPal example inside the dedicated [Examples / Cookbooks](https://shopware-frontends-docs.vercel.app/examples/) section.

**Comment by mkucmus** [comment]
hey guys, the payments topic is expanded in more general way here: https://shopware-frontends-docs.vercel.app/getting-started/payments.html 

let me know if it's kind of what people can rely on :)

**Comment by BrocksiNet** [comment]
For me it is fine 👍 If we want to extend or update we will create a new Issue I would say. 👯
